### ISSUE TITLE
[MultistepDialog] Allow side navigation to be disabled (e.g. when submitting)

### ISSUE THREAD

**Issue by saruman1234** [palantir/blueprint#4872]
#### Environment

- __Package version(s)__: @blueprintjs/core@3.48.0

#### Feature Request

1. Add a `disabled` prop to `<MultistepDialog>` that would allow the side navigation to be disabled (e.g when the dialog is being submitted)

1. Perhaps if specified `true` (and ignored as `false | undefined`), it could also combine & override any individual `disabled` props for all the other steps' `nextButtonProps` and `backButtonProps`? (You can see in my Code Sandbox example, I am passing `|| submitting` through to most of them)

#### Code Sandbox

https://codesandbox.io/s/blueprint-sandbox-multistepdialog-4di0d

#### Steps to reproduce

(Using the example in the Code Sandbox)
1. Advance through each step of the `<MultistepDialog>` to the end.
2. See how each step must be "valid" before you can progress.
3. On the last step, click the "Submit" button to trigger a fake submit state.
4. ❌ See that you can still navigate around between past steps via the side navigation whilst the form submit is ongoing.

**Comment by adidahiya** [comment]
Makes sense. This sounds like a reasonable feature request. We should provide a way to disable the step list in this scenario from your sandbox:

![2021-08-23 14 18 07](https://user-images.githubusercontent.com/723999/130496929-da8635bc-cd84-48df-be46-6a43704a35e0.gif)

**Comment by suwignyo** [comment]
To add to this, is it possible for the left panel to also be hidden completely or would you just suggest using <Dialog /> and doing my own pagination?

**Comment by adidahiya** [comment]
@suwignyo hmm, yeah I suppose I would be open to adding that option. but that would be a separate request, can you file another issue please?
